### PR TITLE
Max Curran HW05 submission

### PR DESCRIPTION
**Pull request by mtcurran**

### CODE REVIEW

**Comment by danielsgriffin** [comment]
Good.
Though:
```python
>>> len("anticonservationist")
19
```

File: HW05_ex00_TextAdventure.py [hunk 1]
```diff
@@ -83,20 +94,24 @@ def dead(why):
 
 
 ##############################################################################
-def start():
+def main():
     # START the TextAdventure game
-    print "You are in a dark room."
-    print "There is a door to your right and left."
-    print "Which one do you take?"
+    user_name = sys.argv[1]
```
Comment (AGagliano) [inline HW05_ex00_TextAdventure.py]: If I don't input my name, the code doesn't run. May want to put a try/except in notifying the user of their mistake.

File: HW05_ex00_TextAdventure.py [hunk 2]
```diff
@@ -2,29 +2,33 @@
...
+        dead("%s implodes." % str(user_name))
         pass
+    elif next == "back":
+        back_room(user_name)
 
 
-def gold_room():
-    print "This room is full of gold.  How much do you take?"
+def gold_room(user_name):
+    print "This room is full of gold.  How much does %s take?" % str(user_name)
```
Comment (AGagliano) [inline HW05_ex00_TextAdventure.py]: I don't believe you need the str() command on user_name if you want to make things easier for yourself.

File: HW05_ex00_TextAdventure.py [hunk 3]
```diff
@@ -33,49 +37,56 @@ def gold_room():
         dead("Man, learn to type a number.")
 
     if how_much < 50:
-        print "Nice, you're not greedy, you win!"
+        print "Nice, %s is not greedy, %s wins!" % (str(user_name), str(user_name))
         exit(0)
     else:
-        dead("You greedy bastard!")
+        print "%s is a greedy bastard! %s is banished to the infinite stairs." % (str(user_name), str(user_name))
+        infinite_stairway_room(user_name, 0)
```
Comment (AGagliano) [inline HW05_ex00_TextAdventure.py]: Haha this is great! But, I think the next iteration, we want to access the infinite_stairway_room() from main() instead.

File: HW05_ex00_TextAdventure.py [hunk 4]
```diff
@@ -33,49 +37,56 @@ def gold_room():
...
+        elif (next == "taunt") and not bear_moved:
+            print "The bear has moved from the door. %s can go through it now." % str(user_name)
             bear_moved = True
-        elif next == "taunt bear" and bear_moved:
-            dead("The bear gets pissed off and chews your leg off.")
-        elif next == "open door" and bear_moved:
-            gold_room()
+        elif (next == "taunt") and bear_moved:
+            dead("The bear gets pissed off and chews %s's leg off." % str(user_name))
+        elif (next == "open" or "door") and bear_moved:
```
Comment (AGagliano) [inline HW05_ex00_TextAdventure.py]: Check this. As the user, if I enter 'door' I end up in the else condition, not this elif condition.

File: HW05_ex00_TextAdventure.py [hunk 5]
```diff
@@ -2,29 +2,33 @@
...
+        back_room(user_name)
 
 
-def gold_room():
-    print "This room is full of gold.  How much do you take?"
+def gold_room(user_name):
+    print "This room is full of gold.  How much does %s take?" % str(user_name)
 
     next = raw_input("> ")
     if "0" in next or "1" in next:
```
Comment (AGagliano) [inline HW05_ex00_TextAdventure.py]: This was the original code, but I don't understand how this works for checking for numbers.

**Pull request closed without merging**